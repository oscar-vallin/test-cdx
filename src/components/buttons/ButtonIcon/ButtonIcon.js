import React from 'react';
import PropTypes from 'prop-types';
import { StyledButton, StyledFontIcon } from './ButtonIcon.styles';

const ButtonIcon = ({ id = '__ButtonIcon', children, disabled = true, icon, size, onClick, ...props }) => {
  return (
    <StyledButton id={id} disabled={disabled} onClick={onClick} {...props}>
      <StyledFontIcon iconName={icon} size={size} />
    </StyledButton>
  );
};

ButtonIcon.propTypes = {
  id: PropTypes.string,
  children: PropTypes.node,
  primary: PropTypes.bool,
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
  icon: PropTypes.string,
};

export { ButtonIcon };
