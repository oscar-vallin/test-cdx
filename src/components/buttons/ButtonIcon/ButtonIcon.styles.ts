import styled from 'styled-components';
import { FontIcon } from '@fluentui/react';
import { Button } from '../Button';

export const StyledButton = styled(Button)`
  color: ${({ theme }) => theme.colors.white};
  background: transparent;
  border: none;
`;

export const StyledFontIcon = styled(FontIcon)<StyledFontIconProps>`
  font-size: ${({ size }) => (size ? `${size}px` : '1.25rem')};
`;

type StyledFontIconProps = {
  size?: number;
};
