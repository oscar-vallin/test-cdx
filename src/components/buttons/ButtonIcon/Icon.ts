import { IIconProps } from 'office-ui-fabric-react';

export const getIcon = () => {
  const addFriendIcon: IIconProps = { iconName: 'AddFriend' };

  return addFriendIcon;
};
