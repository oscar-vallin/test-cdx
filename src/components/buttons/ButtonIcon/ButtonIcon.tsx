import { ReactElement } from 'react';
import { StyledButton, StyledFontIcon } from './ButtonIcon.styles';

const defaultProps = {
  id: '',
  disabled: true,
  text: '',
  icon: '',
  children: '',
  variant: '',
  block: false,
  onClick: () => null,
};

type ButtonIconProps = {
  id?: string;
  disabled?: boolean;
  icon?: string;
  size?: number;
  children?: any;
  text?: string;
  variant?: string;
  onClick?: () => void;
  block?: boolean;
} & typeof defaultProps;

const ButtonIcon = ({ id, disabled = true, icon, size, onClick, ...props }: ButtonIconProps): ReactElement => {
  return (
    <StyledButton id={id} disabled={disabled} onClick={onClick} {...props}>
      <StyledFontIcon iconName={icon} size={size} />
    </StyledButton>
  );
};

ButtonIcon.defaultProps = defaultProps;

export { ButtonIcon };
