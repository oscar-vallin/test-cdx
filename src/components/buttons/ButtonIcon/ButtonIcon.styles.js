import styled from 'styled-components';
import { FontIcon } from 'office-ui-fabric-react/lib/Icon';
import { Button } from '../Button';

export const StyledButton = styled(Button)`
  color: ${({ theme }) => theme.colors.white};
  background: transparent;
  border: none;
`;

export const StyledFontIcon = styled(FontIcon)`
  font-size: ${({ size }) => (size ? `${size}px` : '1.25rem')};
`;
