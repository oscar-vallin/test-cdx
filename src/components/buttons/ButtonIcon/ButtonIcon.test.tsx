import React from 'react';
import { shallow } from 'enzyme';
import { ButtonIcon } from './ButtonIcon.js';

const defaultProps = {
  size: '10',
};

describe('ButtonIcon', () => {
  const mockFn = jest.fn();
  const tree = shallow(
    <ButtonIcon {...defaultProps} icon="edit" onClick={mockFn}>
      Button Contextual
    </ButtonIcon>
  );

  it('Should be defined', () => {
    expect(ButtonIcon).toBeDefined();
  });

  it('Should render correctly', () => {
    expect(tree).toMatchSnapshot();
  });

  it('Should have a click function', () => {
    tree.simulate('click');
    expect(mockFn).toHaveBeenCalled();
  });

  it('Should have a children with the icon prop', () => {
    expect(tree.children().props().iconName).toEqual('edit');
  });
});
