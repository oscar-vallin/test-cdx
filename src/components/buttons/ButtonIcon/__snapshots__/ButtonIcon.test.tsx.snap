// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonIcon Should render correctly 1`] = `ShallowWrapper {}`;
